File: src/main/java/de/nordgedanken/auto_hotkey/psi/AHKFile.java
package de.nordgedanken.auto_hotkey.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import de.nordgedanken.auto_hotkey.lang.core.AhkFileType;
import de.nordgedanken.auto_hotkey.lang.core.AhkLanguage;
import org.jetbrains.annotations.NotNull;

public class AHKFile extends PsiFileBase {
    public AHKFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, AhkLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return AhkFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "AutoHotKey Script";
    }
}

File: src/main/kotlin/de/nordgedanken/auto_hotkey/lang/psi/AhkFile.kt
package de.nordgedanken.auto_hotkey.lang.psi

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import de.nordgedanken.auto_hotkey.lang.core.AhkFileType
import de.nordgedanken.auto_hotkey.lang.core.AhkLanguage

class AhkFile(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, AhkLanguage) {
    override fun getFileType(): FileType = AhkFileType

    override fun toString() = "AutoHotkey Script File"
}
